### DISCUSSION TITLE
Headless only build

### DISCUSSION THREAD

**Discussion by defunctzombie** [cyberbotics/webots#6284] [General]
Is there a way to build webots for headless only mode? i.e. no GUI/Qt stuff required to build and only the simulator and server components to run existing worlds?

**Comment by stefaniapedrazzi** [comment]
No, currently it is not possible to build Webots in headless mode.
But we have some [starting options](https://www.cyberbotics.com/doc/guide/starting-webots#command-line-arguments) to limit the interference of the GUI while running Webots:
* `--batch`: Prevent Webots from creating blocking pop-up windows.
* `--minimize`: Minimize the Webots window on startup.
* `--no-rendering`: Disable rendering in the main 3D view.
